### DISCUSSION TITLE
Extensive logging

### DISCUSSION THREAD

**Discussion by SergeyGulik** [TrakHound/MTConnect.NET#48] [Ideas]
Hi Patrick,

My tests with Adapter, Agent and Client has many simultaneously "moving" parts due to Tasks and MT.
For effective tracing I have introduced some logging which I think may be useful for the product.
Unless you have a reason to avoid logging, I would like to make a PR with my changes.

Best regards,
Sergii.

**Comment by PatrickRitchie** [comment]
Yeah that would be great! I was looking to improve the logging at some point so any help would be appreciated.

Thanks,
-Patrick
